frontend/src/main/java/com/google/tripmeout/frontend/places/PlacesApiPlaceService.java

```diff
@@ -22,14 +19,8 @@ public PlacesApiPlaceService(GeoApiContext context) {
   }
 
   @Override
-  public boolean validatePlaceId(String placesApiPlaceId)
-      throws ApiException, InterruptedException, IOException {
+  public void validatePlaceId(String placesApiPlaceId) throws Exception {
     PlaceDetailsRequest request = new PlaceDetailsRequest(context).placeId(placesApiPlaceId);
-    try {
-      request.await();
-      return true;
-    } catch (InvalidRequestException e) {
-      return false;
-    }
+    request.await();
```

Comment:
Please continue to catch `InvalidRequestException` and translate it into an appropriate domain-specific exception here. The idea is to mask our dependencies/implementation details and have client code only use what it needs. You may want to mint a new exception type if we don't already have something that makes sense for this situation.